### PR TITLE
Report errors to Sentry

### PR DESCRIPTION
**Pull request by andyhd**
## What

* Add dependency on the `raven` Sentry client
* Configure Sentry client with new `SENTRY_DSN` environment variable
* Send errors and uncaught promise rejections to Sentry
* Added `morgan` for HTTP request logging

## How to review

1. Set the `SENTRY_DSN` environment variable
2. Run the app
3. Browse to `/error` - a testing page
4. See an error message displayed
5. Browse to https://sentry.service.dsd.io/mojds/control-panel-frontend/
6. See the error recorded in Sentry

See [Trello #401](https://trello.com/c/ES3ohbQJ)

### CODE REVIEW

File: app/errors.js [hunk 1]
```diff
@@ -1,7 +1,16 @@
...
 
 
+process.on('unhandledRejection', function (error) {
+  log.error(error.message);
+  sentry.captureException(error);
+});
+
+
 module.exports = function (err, req, res, next) {
+  sentry.captureException(err);
```
Comment (xoen) [inline app/errors.js]: How does this line work? Will this send the exception to Sentry straight away?

File: app/errors.js [hunk 2]
```diff
@@ -1,7 +1,16 @@
+var log = require('bole')('error-handler');
+var sentry = require('raven');
 var url_for = require('./routes').url_for;
 
 
+process.on('unhandledRejection', function (error) {
```
Comment (xoen) [inline app/errors.js]: What is this `unhandledRejection` event?

Comment (xoen) [reply app/errors.js]: Oh, [it's happening when a promise's error is not handled](https://nodejs.org/api/process.html#process_event_unhandledrejection).

**Comment by xoen** [comment]
Looks OK to me 👍

File: app/errors.js (hunk 1 above)
Comment (andyhd) [reply app/errors.js]: Yes, this is the global error handler middleware, so it sends the exception to Sentry when an error is passed up the request handler chain.

**Pull request merged**